saleor/graphql/shop/mutations.py

```diff
@@ -60,6 +60,18 @@ class ShopSettingsInput(graphene.InputObjectType):
     customer_set_password_url = graphene.String(
         description="URL of a view where customers can set their password."
     )
+    reserve_stock_duration_anonymous_user = graphene.Int(
+        description=(
+            f"{ADDED_IN_31} Default number of minutes stock will be reserved for "
+            "anonymous checkout or 0 to disable stock reservations."
+        )
+    )
```

Comment:
TBH any int < 1 or null will disable reservations for setting because I'm normalizing those (there are even tests for that).

Reply:
Also if we make this field require we will make partial setting updates through GraphQL API harder.

Reply:
IMO, we should have consistent descriptions. Not all frontend developers know about backend implementation. The description should describe what means all possible values. 
@maarcingebala @krzysztofwolski what do you think?

Reply:
This is robustness principle in practice: be strict about what you send but tolerant in what you receive. I normalize all falsy values (and nonsensical negative ints) to disabled reservations, but to keep things simple in API docs side I don't describe edge cases.

I can reword this to "0 or null to disable".

Reply:
I think we can mention that `0` or `null` disables reservations. I think it's find to make `Int` fields in inputs optional, because as @rafalp said this makes partial updates easier.